Share/Share.Plugin.iOSUnified/ShareImplementation.cs

```diff
@@ -87,10 +93,18 @@ static ShareImplementation()
         /// <param name="text">Text to share</param>
         /// <param name="title">Title of the share popup on Android and Windows, email subject if sharing with mail apps</param>
         /// <param name="excludedActivityTypes">UIActivityType to exclude</param>
-        /// <returns>awaitable Task</returns>
-        public async Task Share(string text, string title = null, params NSString[] excludedActivityTypes)
+        /// <returns>True if the operation was successful, false otherwise</returns>
+        [Obsolete("Use Share(ShareMessage, ShareOptions)")]
+        public Task<bool> Share(string text, string title = null, params NSString[] excludedActivityTypes)
         {
-            await ShareInternal(title, text, null, excludedActivityTypes);
+            var shareMessage = new ShareMessage();
+            shareMessage.Title = title;
+            shareMessage.Text = text;
+
+            var shareOptions = new ShareOptions();
+            shareOptions.ExcludedUIActivityTypes = excludedActivityTypes?.Select(x => (string)x).ToArray();
```

Comment:
My intention behind adding the excluded activity types to the ShareOptions class was to allow the developer to set them from cross platform code and not need platform specific code for this as this defeats the purpose of the lib (at least partially). Of course he would still have to transfer the actual ids from his iOS project to the PCL project, for example using static variables or a platform specific service to obtain the ids. (Obviously if the excluded activities for all share operations are the same he can still use the static list.)

    public class ShareService // this could be PCL code
    {
        public static string[] ExcludedForShareOperationA {get;set;} // set from platform specific code
        public static string[] ExcludedForShareOperationB {get;set;} // set from platform specific code
        
        public void Share()
        {
            if (SomeLogic)
                // share operation A, e.g. a short twitter style message
                Share(msg, new ShareOptions() { ExcludedActivityTypes = ExcludedForShareOperationA });
            else
                // share operation B, e.g. a very long email message
                Share(msg, new ShareOptions() { ExcludedActivityTypes = ExcludedForShareOperationB });
        }
    }